rq/worker.py

```diff
@@ -262,6 +264,26 @@ def __init__(self, queues, name: t.Optional[str] = None, default_result_ttl=DEFA
         elif exception_handlers is not None:
             self.push_exc_handler(exception_handlers)
 
+    def _set_connection(self, connection: Optional['Redis'], default_worker_ttl: int) -> 'Redis':
+        """Improves the reliability of the Worker's connection to the Queue.
+        Will set an ExponentialBackoff logic to the connection error's, and 
+        set a `socket_timout` in case the `BLPOP` command hangs (or any other command).
+
+        Args:
+            connection (Optional[Redis]): The Redis Connection.
+            default_worker_ttl (int): The Default Worker TTL
+        """
+        if connection is None:
+            connection = get_current_connection()    
+        retry = RedisRetry(RedisBackoff(), 3)
+        timeout = default_worker_ttl - 5
+        keepalive = default_worker_ttl - 10
+        connection.socket_timeout = timeout
+        connection.set_retry(retry)
+        connection.socket_connect_timeout = timeout
+        connection.socket_keepalive = keepalive
+        return connection
```

Comment:
Some of these params should be moved to `Redis.connection` attribute.

Reply:
It seems the `Redis` class holds a `connection` attributes, that is being filled with a connection from the `connection_pool`, but every option passed to `Redis` goes directly to the connection pool, so the new version updates the `connection_kwargs` dictionary.